### PR TITLE
Issue 329

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Fix for issue #329

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/azure-service-bus-maintainers this PR is ready for review

File: src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs [hunk 1]
```diff
@@ -63,7 +63,10 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
         {
             if (outgoingMessage.Headers.ContainsKey(Headers.ReplyToAddress))
             {
-                var replyTo = new EntityAddress(outgoingMessage.Headers[Headers.ReplyToAddress]);
+                var replyToAddress = outgoingMessage.Headers[Headers.ReplyToAddress];
+                if (string.IsNullOrWhiteSpace(replyToAddress)) return;
```
Comment (SeanFeldman) [inline src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs]: I'd leave a comment saying "Read-only endpoint case has no reply-to value".

File: src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs [hunk 2]
```diff
@@ -155,7 +158,10 @@ void ApplyDeliveryConstraints(BrokeredMessage brokeredMessage, BatchedOperation
 
         static void CopyHeaders(OutgoingMessage outgoingMessage, BrokeredMessage brokeredMessage)
         {
-            brokeredMessage.Properties[Headers.ReplyToAddress] = brokeredMessage.ReplyTo;
+            if (!string.IsNullOrEmpty(brokeredMessage.ReplyTo))
```
Comment (SeanFeldman) [inline src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs]: So if brokeredMessage.ReplyTo is empty, we don't want even to have a header (empty)?

Comment (yvesgoeleven) [inline src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs]: correct, it serves no purpose

File: src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs (hunk 1 above)
Comment (yvesgoeleven) [inline src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs]: the empty header was actually created by the code on the other comment. But nothing says other causes could exist

**Pull request merged**